provider/aws/aws.go

```diff
@@ -337,7 +356,7 @@ func (p *AWSProvider) records(ctx context.Context, zones map[string]*route53.Hos
 				newEndpoints = append(newEndpoints, endpoint.NewEndpointWithTTL(wildcardUnescape(aws.StringValue(r.Name)), aws.StringValue(r.Type), ttl, targets...))
 			}
 
-			if r.AliasTarget != nil {
+			if r.AliasTarget != nil && aws.StringValue(r.AliasTarget.HostedZoneId) != endpoint.OwnedRecordLabelKey { // see HACK in `newChanges`
```

Comment:
What does this do? Who does the `HostedZoneId` has anything to do with `OwnedRecordLabelKey` itself (`ownedRecord`)?

Also there's no `HACK` anywhere in the file. Is this still needed?

Reply:
Sorry, this was a left-over from the previous implementation approach. Removed.